pp/series_index/querier/regexp/match_analyzer.h

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include "bare_bones/algorithm.h"
+#include "regexp.h"
+
+namespace series_index::querier::regexp {
+
+class RegexpMatchAnalyzer {
+ public:
+  enum class Status : uint8_t {
+    kError = 0,
+    kEmptyMatch,
+    kAnythingMatch,
+    kAllMatch,
+    kAllMatchWithExcludes,
+    kPartialMatch,
+  };
+
+  [[nodiscard]] static Status analyze(re2::Regexp* regexp) {
+    if (!regexp) {
+      return Status::kError;
+    }
+
+    if (regexp->op() == re2::RegexpOp::kRegexpEmptyMatch) {
+      return Status::kEmptyMatch;
+    }
+
+    if (regexp->op() == re2::RegexpOp::kRegexpStar && regexp->sub()[0]->op() == re2::RegexpOp::kRegexpAnyChar) {
+      return Status::kAnythingMatch;
+    }
+
+    if (regexp->op() == re2::RegexpOp::kRegexpPlus && regexp->sub()[0]->op() == re2::RegexpOp::kRegexpAnyChar) {
+      return Status::kAllMatch;
+    }
+
+    if (regexp->op() == re2::RegexpOp::kRegexpConcat) {
+      if (const auto i = skip_begin_text_anchor(regexp); i == skip_end_text_anchor(regexp, i)) {
```

Comment:
It is really hard to read condition for me. Is it true only if there is only one operation between anchors? Is true if there is nothing between anchors? Is true for something like `^^abc[def]?$$$$`?

Reply:
may we extract this code to method w/ explicit name?